src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.cs

```diff
@@ -1161,9 +1160,8 @@ private SolutionCompilationState ComputeFrozenSnapshot(CancellationToken cancell
                     oldDocumentState != newDocumentState &&
                     oldDocumentState.FilePath != newDocumentState.FilePath)
                 {
-                    filePathToDocumentIdsMapChanged = true;
-                    SolutionState.RemoveDocumentFilePath(oldDocumentState, filePathToDocumentIdsMapBuilder);
-                    SolutionState.AddDocumentFilePath(newDocumentState, filePathToDocumentIdsMapBuilder);
+                    filePathToDocumentIdsMapBuilder.Remove(oldDocumentState.FilePath, oldDocumentState.Id);
```

Comment:
>filePathToDocumentIdsMapBuilder.Remove(oldDocumentState.FilePath, oldDocumentState.Id); [](http://example.com/codeflow?start=20&length=87)

Out of curiosity, does the Remove need to happen since this one is frozen?

Reply:
If we didn't remove, we could be in a strange state. The doc-id *is* in the solution, but under a new file path now. But using the old file path would now find it. 

The approach now allows paths to link to docs that are not there. But that's very different from the doc being there, but being in multiple locations.  

I can doc to this effect

Reply:
I guess the way that I'm internalizing this is if the map is frozen, each key's value (the document id array) is suspect. If that's the case, then we could consider only doing an Add if the dictionary doesn't contain the key already.

Reply:
I'm not sure I understand exactly. W can chat tomorrow though!

Reply:
Nope, this was me trying to review some code when I was tired. Ignore this comment thread.